oracle.go

```diff
@@ -280,32 +281,36 @@ func (o *Oracle) RotateRootCredentials(ctx context.Context, statements []string)
 	return o.RawConfig, nil
 }
 
-func (o *Oracle) disconnectSession(db *sql.DB, username string) error {
+func (o *Oracle) disconnectSession(db *sql.DB, username string) (rerr error) {
 	disconnectStmt, err := db.Prepare(strings.Replace(sessionQuerySQL, "{{name}}", username, -1))
 	if err != nil {
-		return err
+		return multierror.Append(rerr, err)
 	}
 	defer disconnectStmt.Close()
 	if rows, err := disconnectStmt.Query(); err != nil {
-		return err
+		return multierror.Append(rerr, err)
 	} else {
-		defer rows.Close()
 		for rows.Next() {
 			var sessionID, serialNumber int
 			var username sql.NullString
 			err = rows.Scan(&sessionID, &serialNumber, &username)
 			if err != nil {
-				return err
-			}
-			killStatement := fmt.Sprintf(sessionKillSQL, sessionID, serialNumber)
-			_, err = db.Exec(killStatement)
-			if err != nil {
-				return err
+				rows.Close()
+				return multierror.Append(rerr, err)
 			}
+			// Issuing the KILL SESSION must happen *after* the rows cursor is closed.
+			defer func() {
+				killStatement := fmt.Sprintf(sessionKillSQL, sessionID, serialNumber)
+				_, err = db.Exec(killStatement)
+				if err != nil {
+					rerr = multierror.Append(rerr, err)
+				}
+			}()
 		}
+		defer rows.Close()
 		err = rows.Err()
 		if err != nil {
-			return err
+			return multierror.Append(rerr, err)
```

Comment:
And here

Reply:
Thanks. I wasn't sure what the best practice here was. I'll change those back.